buildenv/jenkins/JenkinsfileBase

```diff
@@ -967,13 +974,44 @@ def addGrinderLink() {
 		if ( key == "LABEL" ) {
 			labelValue = "LABEL=${value}"
 		}
+		if ( key == "TARGET" ) {
+			targetValue = "TARGET=${value}"
+		}
 	}
+	env.RERUN_LINK = url
+	env.FAILED_TEST_TARGET = targetValue
+	currentBuild.description += "<br><a href=\"https://github.com/adoptium/aqa-tests/wiki/How-to-Run-a-Grinder-Build-on-Jenkins\">Grinder Wiki</a>"
 	echo "Rerun in Grinder: ${url}"
 	currentBuild.description += "<br><a href=${url}>Rerun in Grinder</a> Change TARGET to run only the failed test targets."
 	def sameMachineUrl = url.replace(labelValue,"LABEL=${NODE_NAME}")
 	echo "Rerun in Grinder on same machine: ${sameMachineUrl}"
 	currentBuild.description += "<br><a href=${sameMachineUrl}>Rerun in Grinder on same machine</a> Change TARGET to run only the failed test targets. For dynamic vm agents as they are created at runtime and quickly recycled, it may not be possible to rerun on the same agent."
-	currentBuild.description += "<br><a href=\"https://github.com/adoptium/aqa-tests/wiki/How-to-Run-a-Grinder-Build-on-Jenkins\">Grinder Wiki</a>"
+}
+
+def addFailedTestsGrinderLink(){
+	if (currentBuild.result == 'UNSTABLE' || currentBuild.result == 'FAILURE' || currentBuild.result == 'ABORTED') {
+		echo "Fetching Tap files..."
+		def tapFiles = findFiles(glob: "**/*.tap")
+		def failedTestList = "testList+TESTLIST="
+		for (def tapFile: tapFiles) {
+			echo "Tap file found: ${tapFile}"
+			def file = readFile(file: "${tapFile}")
+			TapConsumer tapConsumer = TapConsumerFactory.makeTap13Consumer()
+			TestSet testSet = tapConsumer.load(file)
+			for (int i = 1; i <= testSet.getNumberOfTestResults(); i++) {
+				if (!testSet.getTestResult(i).getStatus().toString().equals("ok")) {
+					def testCase = testSet.getTestResult(i).getDescription().substring(2)
+					failedTestList += "$testCase,"
+				}
+			}
+		}
+		failedTestList = failedTestList.substring(0, failedTestList.length() - 1)
+		echo "Fail test list is ${failedTestList}"
+		String url = env.RERUN_LINK
+		def failedTestUrl = url.replace(env.FAILED_TEST_TARGET, "TARGET=$failedTestList")
+		echo "Rerun failed tests in Grinder: $failedTestList"
+		currentBuild.description += "<br><a href=${failedTestUrl}> Rerun in Grinder with failed tests targets</a>"
```

Comment:
Please keep line 1012 and 1013 message to be the same - `Rerun in Grinder with failed test targets ` and the rerun  link (not just test list)